operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,628 @@
+                            .into_iter()
+                            .map(num_traits::AsPrimitive::as_)
+                            .collect::<Vec<f32>>()
+                    })
+                    .collect();
+
+                // let pixels: Vec<_> = rasters.into_iter().collect();
+                let model_content = std::fs::read_to_string(model).unwrap();
+
+                process_tile(
+                    &pixels,
+                    &pool,
+                    model_content.as_bytes(),
+                    grid_shape,
+                    n_bands as usize,
+                    ndv.unwrap(),
+                )
+            })
+            .await
+            .unwrap();
+
+        let rt: BaseTile<GridOrEmpty<GridShape<[usize; 2]>, f32>> =
+            RasterTile2D::new_with_properties(
+                time,
+                tile_position,
+                global_geo_transform,
+                predicted_grid.into(),
+                props.clone(),
+            );
+
+        Ok(rt)
+    }
+}
+
+fn process_tile(
+    bands_of_tile: &Vec<f32>,
+    pool: &ThreadPool,
+    model_file: &[u8],
+    grid_shape: GridShape<[usize; 2]>,
+    n_bands: usize,
+    nan_val: f32,
+) -> geoengine_datatypes::raster::Grid<GridShape<[usize; 2]>, f32> {
+    pool.install(|| {
+        // to get one row of data means taking (n_pixels * n_bands) elements
+        let chunk_size = grid_shape.number_of_elements() * n_bands;
+
+        let res: Vec<_> = bands_of_tile
+            .par_chunks(chunk_size)
+            .map(|elem| {
+                // get xgboost style matrices
+                let xg_matrix = match DMatrix::from_col_major_f32(
+                    elem,
+                    mem::size_of::<f32>() * n_bands,
+                    mem::size_of::<f32>(),
+                    grid_shape.number_of_elements(),
+                    n_bands,
+                    -1,
+                    nan_val,
+                ) {
+                    Ok(matrix) => matrix,
+                    Err(err) => panic!(
+                        "XGBoost failed to create DMatrix with error msg: \n{:?}",
+                        err
+                    ),
+                };
+
+                let mut out_dim: u64 = 0;
+                let bst = match Booster::load_buffer(model_file) {
+                    Ok(booster) => booster,
+                    Err(err) => panic!(
+                        "XGBoost failed to load model from file with error msg: \n{:?}",
+                        err
+                    ),
+                };
+
+                // measure time for prediction
+                match bst.predict_from_dmat(
+                    &xg_matrix,
+                    &[grid_shape.number_of_elements() as u64, n_bands as u64],
+                    &mut out_dim,
+                ) {
+                    Ok(res) => res,
+                    Err(err) => {
+                        panic!("XGBoost failed to predict data with error msg: \n{:?}", err)
```

Comment:
return error instead of panic

Reply:
https://github.com/geo-engine/geoengine/pull/639/commits/84fb951c583cb7141ff50f1196f90504d2ef8a90